src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -13757,5 +13757,92 @@ public static void Main()
 $@"{s_expressionOfTDelegate1ArgTypeName}[<>f__AnonymousDelegate0]
 {s_expressionOfTDelegate1ArgTypeName}[System.Func`2[System.Int32,<>f__AnonymousDelegate0]]").VerifyDiagnostics();
         }
+
+        [Fact]
+        public void ParamsArray_SynthesizedTypesMatch()
+        {
+            var source = """
+                static void Report(object obj1, object obj2) => System.Console.WriteLine($"{obj1.GetType() == obj2.GetType()}, {obj1.GetType()}");
+
+                var lam1 = (params int[] xs) => xs.Length;
+                int Method1(params int[] xs) => xs.Length;
+                var del1 = Method1;
+                Report(lam1, del1);
+
+                var lam2 = (params int[] ys) => ys.Length;
+                int Method2(params int[] ys) => ys.Length;
+                var del2 = Method2;
+                Report(lam2, del2);
```

Comment:
>Report(lam2, del2) [](http://example.com/codeflow?start=16&length=18)

Is this case different from `Report(lam1, del1)`?

Reply:
> Is this case different from `Report(lam1, del1)`?

Slightly, it has different parameter names (`xs` vs `ys`).

> If the intent is to check that multiple lambdas may have the same synthesized delegate

You're right, that was the intention.

> perhaps also add `Report(lam1, lam2);`.

Added: b68a0142aca7f8ae149cfad019e75425f3faa316